app/models/core.ts

```diff
@@ -79,7 +80,8 @@ export class Graph extends Element {
     exampleTuple = ["hello", 10];
 
     constructor(public plugin: Plugin) {
-        super(plugin.graphPluginData());
+        // TODO: fix that this is obviously special cased
+        super(plugin.graphPluginData("DFAGraph"));
```

Comment:
It seems like Element could be an interface instead of a class. I'm not seeing an explosion of duplicate code by making Element an interface.

Personally, I think that the inheritance is a bigger issue than the annotations. I just think that the two combined spells for complicated code. Imagine Java Spring combined with Swing.